### ISSUE TITLE
Omega(n,q) is not always a subgroup of SO(n,q)

### ISSUE THREAD

**Issue by dimpase** [gap-system/gap#500]
E.g.

```
gap> G:=SO(3,9);
SO(0,3,9)
gap> G1:=Omega(3,9);
Omega(0,3,9)
gap> G1.1 in G;
false
```

This contradicts documentation (50.2-8), which says "This group has always index two in SO( e, d, q )". 
(So it stabilises a different form)...

**Comment by dimpase** [comment]
Specifically, in this case, in coordinates X,Y,Z, the group SO preserves the quadratic form XY-Z^2, and Omega preserves the form XZ-Y^2.

**Comment by stevelinton** [comment]
Section 50.2 of the documentation explains where all the generators come from and specifically says "Note  that  due  to  the different sources for the generators, the invariant
  forms  for the groups ohm(e,d,q) are in general different from the forms for
  SO(e,d,q) and GO(e,d,q)."

So the fix is probably just to amend 50.2-8.

**Comment by fingolfin** [comment]
It would be kind of nice if they used the same form, though. Achieving that shouldn't be very hard. I can look into it (maybe not today, but...).

**Comment by fingolfin** [comment]
The only concern is: Might there be existing code which breaks if we change which form is used?

**Comment by dimpase** [comment]
one can make a new function, called SOmega, say, and put a new
implementation there. And deprecate Omega.

**Comment by frankluebeck** [comment]
It is maybe good to have generators which can be found in the literature available in GAP. 

If someone wants to look at this: It would be nice to store the bilinear/quadratic form in the group returned by `Omega` similar to what is stored in the group by `SO` or `Sp`.

Could one use the `Forms` package to provide a variant of `SO`, `Omega`, `Sp` and so on where the form is given as an argument?

**Comment by dimpase** [comment]
Forms package is broken in characteristic 2, at least. Try using it on SO(3,2)... (This bug has been discovered today by me and reported to the author...)

**Comment by fingolfin** [comment]
Using the forms package to provide such variants is actually quite easy (modulo bugs, such as the one reported by Dima, of course). Here is an example:
```
gap> G:=Omega(3,3);
Omega(0,3,3)
gap> formOmega := PreservedSesquilinearForms(G)[1];
< bilinear form >
gap> Display(formOmega);
Bilinear form
Gram Matrix:
 . . 1
 . 2 .
 1 . .
gap> H:=GO(3,3);
GO(0,3,3)
gap> formGO := PreservedSesquilinearForms(H)[1];
< bilinear form >
gap> Display(formGO);
Bilinear form
Gram Matrix:
 . 2 .
 2 . .
 . . 1
gap> b := BaseChangeToCanonical(formGO)^-1 * BaseChangeToCanonical(formOmega);
[ [ Z(3), Z(3)^0, Z(3)^0 ], [ Z(3)^0, 0*Z(3), 0*Z(3) ], [ Z(3)^0, Z(3)^0, 0*Z(3) ] ]
gap> Display(b * GramMatrix(formOmega) * TransposedMat(b));
 . 1 .
 1 . .
 . . 2
gap> K := Group(List(GeneratorsOfGroup(G), g -> g^(b^-1)));;
gap> IsSubgroup(H, K);
true
```

**Comment by hulpke** [comment]
> It is maybe good to have generators which can be found in the literature available in GAP.

In view of the project to port the composition tree from Magma to GAP (and of people using both systems) it also might be useful to have compatibility with Magma.

**Comment by fingolfin** [comment]
So with the above example, one could have methods `GO(form)`, `Omega(form)` etc.

**Comment by fingolfin** [comment]
BTW, see also https://bitbucket.org/jdebeule/forms/issues/1/preservedsesquilinerforms

**Comment by alex-konovalov** [comment]
PR #503 addressed this and have been merged - may we close this now? Is there anything left?

**Comment by dimpase** [comment]
I don't see why this can be closed - yes, a doc fixhas been merged, to document this discrepancy. But it is still a discrepancy.

**Comment by stevelinton** [comment]
Someone, perhaps accidentally made a choice to follow specific papers rather than use a common form.
For backwards compatibility we are not likely to change that choice without a better reason that "tidyness". 

So we are left with the idea of an extension in the forms package (or a second package dependent on forms) to allow construction of these groups with arbitrary forms, which seems like a nice idea, so I guess the issue should stay open, perhaps renamed to preserve that idea.

**Comment by fingolfin** [comment]
PR #4489 was merged. Together with forms 1.2.7 (hopefully will be released soon -- 1.2.6 was meant to do this, but unfortunately we discovered an issue after release that breaks it, but a fix is being processed right now).

From my POV, that PR resolves this issue here completely.
